### DISCUSSION TITLE
is react hook form the right tool for my case ?

### DISCUSSION THREAD

**Discussion by SuperChuck83** [react-hook-form/react-hook-form#3637] [Q&A]
Hello, 
i need to create CRUD ( invoice crud for example) with complex object. 

For example my invoice is an object with 100 parameters and i get it from server, in my crud i'm using only 20 inputs.
so i created  4 components=>**4 form interfaces ( customer, item, ... )** with 5 parameters, 

then every time i detect an update on one input on my forms ( with useWatch ) i update my complex invoice object (with **recoil**) 
then when user push save button i send the complex object to the server. 

is it the right way to do ? 
is react hook form the right tool for my case ? 

Thx

**Comment by bluebill1049** [comment]
hey @SuperChuck83 i don't have a definite answer, I think it's to have a play with all the tools before pick one which will work for your usecase.
